File: MaquinaTicket/MaquinaTicket/MaquinaTicket.cs
﻿using System;

/*
 * AUTOR=LUIS ALFREDO SUAREZ LITARDO
 * 2017-06-03
 * CLASE MAQUINA DE TICKETS
 * TELEMÁTICA 4TO MÓDULO
 */
namespace MaquinaTicket
{
	public class MaquinaTicket
	{

		private int precio;
		private int balance;
		private int total;
		private int contadorTicket;

		public MaquinaTicket (int ticketCost)
		{
			if (ticketCost>0) {
				Precio = ticketCost;
				Balance = 0;
				total = 0;
				ContadorTicket = 0;
			} else {
				Console.WriteLine ("Inserte una cantidad mayor a 0");
			}

		}
		//Obtener Precio Y Setear Precio
		public int Precio
		{
			get{ 
				return precio;
			}
			set{ 
				if (value > 0) {
					precio = value;
				}
			}
		}
		//Obtener Balance y setear balance
		public int Balance
		{
			get{ 
				return balance;
			}
			set{ 
				if (value >= 0) {
					balance = value;
				} else {
					Console.WriteLine ("Inserte una Moneda Mayor a 0");	
				}
			}
		}
		public int ContadorTicket{
			get{ 
				return contadorTicket;
			}
			set { 
				if (value >= 0) {
					contadorTicket = value;
				}
			}
		}
		public int Total{
			get{ 
				return total;
			}
			set{ 
				if (total >= 0) {
					total = value;
				}
			}
		}
		//Vaciar Maquina y retornar el total
		public int VaciarMaquina{
			get{
				int TempTotal = total;
				Total = 0;
				ContadorTicket = 0;
				return TempTotal;
			}
		}

		public int reembolsoBalance{
			get	
			{
				int montoRetorno = Balance;
				Balance = 0;
				return montoRetorno;
			}
		}
		//Imprimir ticket
		public void ImprimirTicket(){
			if (Balance>=Precio) {
				ContadorTicket++;
				Console.WriteLine ("--------------------------------");
				Console.WriteLine ("TICKET  visite:www.maquinaticket.com");
				Console.WriteLine ("--------------------------------");
				Console.WriteLine ("TICKET #{0}",ContadorTicket);
				Console.WriteLine ("PRECIO --> {0} centavos",Precio);
				Console.WriteLine ("--------------------------------");
				Total += Precio;
				Balance -= Precio;

			} else {
				Console.WriteLine ("Usted necesita {0} centavos mas",(Precio-Balance));
			}
		}
   }
}

File: python/MaquinaTicket.py
class MaquinaTicket:
    """docstring for MaquinaTicket."""
    def __init__(self,precio=0,balance=0,total=0,contadorTicket=0):
        if precio>0:
            self.__precio = precio
            self.__balance = balance
            self.__contadorTicket = contadorTicket
            self.__total = total
        else:
            print "Inserte un precio mayor a 0"

    """METODOS DE LA CLASE"""
    def getPrecio(self):
        return self.__precio

    def getBalance(self):
        return self.__balance

    def InsertarMoneda(self,moneda=0):
        if moneda>0:
            self.__balance += moneda
        else:
            print "Inserte Moneda mayor a 0"


    def VaciarMaquina(self):
        TotalTemp = self.__total
        self.__total = 0
        return TotalTemp

    def ImprimirTicket(self):
        if self.__balance >= self.__precio:
            self.__contadorTicket += 1
            print "----------------"
            print "TICKET # %d" % self.__contadorTicket
            print "PRECIO = %d" % self.__precio
            print "----------------"
            self.__total += self.__precio
            self.__balance -= self.__precio
        else:
            print "Necesita %d centavos mas" % (self.__precio - self.__balance)









objeto = MaquinaTicket(10)
objeto.InsertarMoneda(50)

print objeto.getPrecio()
